src/handlers/guild.ts

```diff
@@ -622,3 +622,100 @@ export function getGuild(guildID: string, counts = true) {
     { with_counts: counts },
   ) as Promise<UpdateGuildPayload>;
 }
+
+/**
+ * Returns the guild template if it exists
+ * @param guildID The ID of the guild
+ * @param code The code of the template to get
+ */
+export function getGuildTemplate(guildID: string, code: string) {
+  const endpoint = `${endpoints.GUILD_TEMPLATES(guildID)}/${code}`;
+  return RequestManager.get(endpoint);
+}
+
+/**
+ * Create a new guild based on a template
+ * NOTE: This endpoint can be used only by bots in less than 10 guilds.
+ * @param code The code of the template to create a guild from
+ */
+export function createGuildFromTemplate(
+  code: string,
+  data: CreateGuildFromTemplate,
+) {
+  return RequestManager.post(endpoints.GUILD_TEMPLATE(code), data);
+}
+
+export interface CreateGuildFromTemplate {
+  /** name of the guild (2-100 characters) */
+  name: string;
+  /** base64 128x128 image for the guild icon */
+  icon?: string;
+}
+
+/**
+ * Returns an array of guild templates
+ * @param guildID The ID of the guild
+ */
+export function getGuildTemplates(guildID: string) {
+  return RequestManager.get(endpoints.GUILD_TEMPLATES(guildID));
+}
+
+/**
+ * Deletes a template from a guild
+ * @param guildID The guild ID to delete the template from
+ */
+export function deleteGuildTemplate(guildID: string, code: string) {
+  const endpoint = `${endpoints.GUILD_TEMPLATES(guildID)}/${code}`;
+  return RequestManager.delete(endpoint);
+}
+
+/**
+ * Creates a template for the guild
+ * @param guildID The ID of the guild to create a template in
+ * @param name name of the template (1-100 characters)
+ * @param description description for the template (0-120 characters
+ */
+export function createGuildTemplate(
+  guildID: string,
+  data: CreateGuildTemplate,
+) {
+  return RequestManager.post(endpoints.GUILD_TEMPLATES(guildID), data);
+}
+
+export interface CreateGuildTemplate {
+  /** name of the template (1-100 characters) */
+  name: string;
+  /** description for the template (0-120 characters) */
+  description?: string | null;
+}
+
+/**
+ * Syncs the template to the guild's current state
+ * @param guildID The ID of the guild
+ * @param code The code of the template to sync
+ */
+export function syncGuildTemplate(guildID: string, code: string) {
+  const endpoint = `${endpoints.GUILD_TEMPLATES(guildID)}/${code}`;
+  return RequestManager.put(endpoint);
+}
+
+/**
+ * Edit a template's metadata
+ * @param guildID The ID of the guild
+ * @param code The code of the template to edit
+ */
+export function editGuildTemplate(
+  guildID: string,
+  code: string,
+  data: EditGuildTemplate,
+) {
+  const endpoint = `${endpoints.GUILD_TEMPLATES(guildID)}/${code}`;
```

Comment:
same move this endpoint.

Add checks for name and description character lengths